fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/QuantifiedApply2JoinRule.java

```diff
@@ -51,27 +53,51 @@ public boolean check(OptExpression input, OptimizerContext context) {
                 && !SubqueryUtils.containsCorrelationSubquery(input);
     }
 
+    private static ScalarOperator normalizeBinaryOperator(BinaryPredicateOperator input) {
+        if (input.getUsedColumns().isEmpty()) {
+            return new FoldConstantsRule().visitBinaryPredicate(input, null);
+        } else if (input.getChild(0).getUsedColumns().isEmpty()) {
+            // normalize binaryPredicateOperator, e.g. 1 = tbl.v1 to tbl.v1 = 1
+            input.swap();
+        }
+        return input;
+    }
+
     @Override
     public List<OptExpression> transform(OptExpression input, OptimizerContext context) {
         LogicalApplyOperator apply = (LogicalApplyOperator) input.getOp();
+        boolean isNotIn = false;
+        ScalarOperator simplifiedPredicate = null;
+        if (apply.getSubqueryOperator() instanceof MultiInPredicateOperator) {
+            MultiInPredicateOperator multiIn = (MultiInPredicateOperator) apply.getSubqueryOperator();
+            isNotIn = multiIn.isNotIn();
+            List<ScalarOperator> conjuncts = Lists.newArrayList();
+            ScalarOperatorRewriter rewriter = new ScalarOperatorRewriter();
+            for (int i = 0; i < multiIn.getTupleSize(); ++i) {
+                ScalarOperator left = multiIn.getChild(i);
+                ScalarOperator right = multiIn.getChild(multiIn.getTupleSize() + i);
+                ScalarOperator conjunct =
+                        normalizeBinaryOperator(new BinaryPredicateOperator(BinaryPredicateOperator.BinaryType.EQ,
+                                left, right));
+                ScalarOperator conjunctWithCast =
+                        rewriter.rewrite(conjunct, ScalarOperatorRewriter.DEFAULT_TYPE_CAST_RULE);
```

Comment:
do cast with implicitCastRule plz, cast affects correctness, it's best to deal with it early